content/docs/reference-react-component.md

```diff
@@ -274,22 +275,22 @@ Currently, if `shouldComponentUpdate()` returns `false`, then [`UNSAFE_component
 static getDerivedStateFromProps(props, state)
 ```
 
-`getDerivedStateFromProps` is invoked right before calling the render method, both on the initial mount and on subsequent updates. It should return an object to update the state, or null to update nothing.
+המתודה `getDerivedStateFromProps` מתבצעת רגע לפני הקריאה למתודת הרינדור, גם ב-mount הראשוני וגם על עדכונים מאוחרים. עליה להחזיר אובייקט לעדכן את ה-state, או null כדי לעדכן כלום.
 
-This method exists for [rare use cases](/blog/2018/06/07/you-probably-dont-need-derived-state.html#when-to-use-derived-state) where the state depends on changes in props over time. For example, it might be handy for implementing a `<Transition>` component that compares its previous and next children to decide which of them to animate in and out.
+המתודה הזו קיימת [לשימושים נדירים](/blog/2018/06/07/you-probably-dont-need-derived-state.html#when-to-use-derived-state) בהם ה-state תלוי בשינויים ב-props במהלך הזמן. לדוגמא, ייתכן שיהיה זה שימושי לשימוש במימוש של קומפוננטת `<Transition>` שמשווה את הילדים הקודמים והבאים שלה כדי להחליט איזה מהם להנפיש.
 
-Deriving state leads to verbose code and makes your components difficult to think about.  
-[Make sure you're familiar with simpler alternatives:](/blog/2018/06/07/you-probably-dont-need-derived-state.html)
+שימוש ב- deriving state מוביל לקוד ארוך ומקשה על חשיבה על הקומפוננטות שלכם.
+[ודאו שאתם מכירים מימוש עם אלטרנטיבות פשוטות יותר:](/blog/2018/06/07/you-probably-dont-need-derived-state.html)
 
-* If you need to **perform a side effect** (for example, data fetching or an animation) in response to a change in props, use [`componentDidUpdate`](#componentdidupdate) lifecycle instead.
+* אם אתם צריכים **ליצור side-effect** (לדוגמא, קבלת מידע או הנפשה) כתגובה לשינוי ב-props, השתמשו ב- [`componentDidUpdate`](#componentdidupdate) מעגל-החיים במקום.
 
-* If you want to **re-compute some data only when a prop changes**, [use a memoization helper instead](/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization).
+* אם אתם רוצים **לחשב מחדש מידע מסוים רק כאשר prop משתנה**, [השתמשו ב-memoization helper במקום](/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization).
 
-* If you want to **"reset" some state when a prop changes**, consider either making a component [fully controlled](/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-controlled-component) or [fully uncontrolled with a `key`](/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key) instead.
+* אם אתם רוצים **לאפס state כלשהו כאשר prop משתנה**, שקלו לייצר קומפוננטה [fully controlled](/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-controlled-component) או [fully uncontrolled with a `key`](/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key) במקום.
```

Comment:
```suggestion
* אם אתם רוצים **לאפס state כלשהו כאשר prop משתנה**, שקלו לייצר קומפוננטה [נשלטת לגמרי](/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-controlled-component) או [לא נשלטת עם `key`](/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key) במקום.
```
You forgot to translate the link text

Reply:
actually, it was intentionally kept that way :) 
in my opinion "fully controlled/uncontrolled" are terms that don't have decent alternative in Hebrew.